apps/ws/src/Game.ts

```diff
@@ -30,46 +31,56 @@ export function isPromoting(chess: Chess, from: Square, to: Square) {
       .includes(to);
 }
 
+export interface Player {
+    id?: string
+    userId: string
+    name: string
+}
+
 export class Game {
     public gameId: string;
-    public player1UserId: string;
-    public player2UserId: string | null;
+    public player1: Player;
+    public player2: Player | null;
     public board: Chess
     private startTime: Date;
     private moveCount = 0;
 
-    constructor(player1UserId: string, player2UserId: string | null) {
-        this.player1UserId = player1UserId;
-        this.player2UserId = player2UserId;
+    constructor(player1: Player, player2: Player | null) {
+        this.player1 = player1;
+        this.player2 = player2;
         this.board = new Chess();
         this.startTime = new Date();
         this.gameId = randomUUID();
     }
 
-    async updateSecondPlayer(player2UserId: string) {
-        this.player2UserId = player2UserId;
-        
-        const users = await db.user.findMany({
-            where: {
-                id: {
-                    in: [this.player1UserId, this.player2UserId ?? ""]
-                }
-            }
-        });
+    isGuestGame() {
+        if (isGuest(this.player1) || isGuest(this.player2)) {
+            return true
+        }
+        return false
+    }
 
-        try {
-            await this.createGameInDb(); 
-        } catch(e) {
-            console.error(e)
-            return;
+    async updateSecondPlayer(player2: Player) {
+        this.player2 = player2;
+        let player1Name = this.player1.name
+        let player2Name = this.player2?.name
+
+        // only create the game in db if its not a guest game
+        if (!this.isGuestGame()) {
```

Comment:
Still needs to be in Db
1. create a new table called guests
2. Can we somehow have an `or` on foreign keys from Game  (whiteplayer, blackPlayer) to Users | Guests (answer could totally be no)

Reply:
Maybe we can have a GuestGame table which points to GuestMoves table?